packages/ia-pic-uploader/src/ia-pic-uploader.ts

```diff
@@ -368,70 +351,81 @@ export class IAPicUploader extends LitElement {
 
     // get input file
     const inputFile = this.fileSelector?.files[0];
-    const getParams = `identifier=${this.identifier}&fname=${encodeURIComponent(
-      inputFile.name,
-    )}&submit=1`;
-
-    await BackendServiceHandler({
-      action: 'save-file',
-      identifier: this.identifier,
-      file: inputFile,
-      getParam: getParams,
-      endpoint: this.endpoint,
-      headers: { 'Content-type': 'multipart/form-data; charset=UTF-8' },
-      callback: async () => {
-        log('callback invoked!', this.type);
-        if (this.type === 'full') await this.metadataAPIExecution();
-      },
-    });
 
-    this.dispatchEvent(new Event('fileUploaded'));
-    if (this.type === 'compact') this.showLoadingIndicator = false;
+    const formData = new FormData();
+    formData.append('file', inputFile);
 
-    //  clear file input
-    if (this.fileSelector) this.fileSelector.value = '';
+    try {
+      const saveResponse = await fetch(
+        `${this.postFileServiceUrl}&fname=${encodeURIComponent(inputFile.name)}`,
+        {
+          method: 'POST',
+          body: formData,
+          credentials: 'include',
+        },
+      );
+      log('saveResponse', saveResponse);
+
+      if (saveResponse) {
+        this.dispatchEvent(new Event('fileUploaded'));
+        if (this.type === 'full') await this.metadataAPIExecution();
+        log('file saved, metadata call started');
+      } else {
+        log('Failed to save file', saveResponse);
+      }
+    } catch (err) {
+      log('Failed to save file', err);
+      this.showLoadingIndicator = false;
+    } finally {
+      if (this.type === 'compact') this.showLoadingIndicator = false;
+
+      // clear file input
+      if (this.fileSelector) this.fileSelector.value = '';
+    }
   }
 
   /**
    * after upload, verify using metadata API if successfully uploaded or not
-   *
-   * @memberof IAPicUploader
    */
-  metadataAPIExecution() {
+  async metadataAPIExecution() {
     const now = Math.round(Date.now() / 1000); // like unix time()
 
     const metadataApiInterval = setInterval(async () => {
```

Comment:
I'm not familiar with how the uploader works, but what is this `setInterval()` doing?